app/controllers/pieces_controller.rb

```diff
@@ -7,19 +7,19 @@ def update
       return render plain: 'Forbidden :(', status: :forbidden
     end
 
-    current_piece.move_to!(params[:piece][:file], params[:piece][:rank])
-
-    if current_piece.valid?
+    if current_piece.valid_move?(params[:piece][:file].to_i, params[:piece][:rank].to_i)
+      current_piece.move_to!(params[:piece][:file].to_i, params[:piece][:rank].to_i)
```

Comment:
How about pulling `.valid_move?` inside `.move_to!`  In:
```
def move_to!(new_col, new_row)
  return unless valid_move?(new_col, new_row)
  etc
```

Reply:
The only problem that I foresee doing that is move_to! doesn't return a true/false so there wouldn't be a way to tell what's a valid_move in the pieces controller unless I'm not seeing something obvious.

Reply:
I was thinking that when calling `move_to!` the first thing it should check is that a move is valid? Since the method would return/interrupt the flow if `valid_move?` is false, then the update should not happen.
We could write `return false unless valid_move?(new_col, new_row)` instead?